app/src/pages/inside/testCaseLibraryPage/importTestCaseModal/importTestCaseModal.tsx

```diff
@@ -16,24 +25,97 @@ import styles from './importTestCaseModal.scss';
 
 export const IMPORT_TEST_CASE_MODAL_KEY = 'importTestCaseModalKey';
 
-const MAX_FILE_SIZE_MB = 50;
+const MAX_FILE_SIZE_MB = 10;
+const MAX_FILE_SIZE_BYTES = MAX_FILE_SIZE_MB * 1024 * 1024;
+const MIME_PART = MIME_TYPES as unknown as {
+  csv: MimeType;
+  xls: MimeType;
+  plain: MimeType;
+};
+const CSV_MIME_TYPES = [MIME_PART.csv, MIME_PART.xls, MIME_PART.plain] as const;
 
 const cx = createClassnames(styles);
 
+type FileLike = File | { file: File };
+type FileInput = FileLike | FileLike[];
+
+const extractFolderIdFromHash = (hash: string) => {
+  const m = hash.match(/\/testLibrary\/folder\/(\d+)/i);
+  return m ? Number(m[1]) : undefined;
+};
+
+const toMB = (bytes: number) => +(bytes / (1024 * 1024)).toFixed(2);
+
 export const ImportTestCaseModal = () => {
   const { formatMessage } = useIntl();
   const [folderName, setFolderName] = useState('');
+  const [file, setFile] = useState<File | null>(null);
+  const [folderIdFromUrl, setFolderIdFromUrl] = useState<number | undefined>(() =>
+    extractFolderIdFromHash(window.location.hash),
+  );
+  const { isImportingTestCases, importTestCases } = useImportTestCase();
+
+  useEffect(() => {
+    const onHashChange = () => setFolderIdFromUrl(extractFolderIdFromHash(window.location.hash));
+    window.addEventListener('hashchange', onHashChange);
+    return () => window.removeEventListener('hashchange', onHashChange);
+  }, []);
 
+  const handleImport = () => {
+    if (!file) {
+      return;
+    }
+    if (folderIdFromUrl != null) {
+      return importTestCases({
+        file,
+        testFolderId: folderIdFromUrl,
+      });
+    }
+    if (!folderName.trim()) {
```

Comment:
Let's split conditions by empty line
```suggestion
    }

    if (!folderName.trim()) {
```